app/models/incoming_email.rb

```diff
@@ -6,6 +6,12 @@ class IncomingEmail < ActiveRecord::Base
   belongs_to :post
   belongs_to :group, foreign_key: :imap_group_id, class_name: 'Group'
 
+  before_save do
+    if self.created_via.blank?
```

Comment:
I am not sure why I need to do this even though I added a default value for the column? If I remove this hundreds of tests fail. cc @tgxworld

Reply:
This is because we have `NULL` set to `false`. Therefore, we are not allowed to pass a `NULL` value when inserting the row. `default` works only when a column is nullable. If `NULL` is passed to the column, the value of the column will be that of the `default` value. 

For the `created_via` column, I don't think we want to automatically populate it via a callback. Instead, we can add a `presence: true` validation so that anyone inserting a new row needs to pass in the `created_via` value. This will prevent any future changes from having an "unknown" `created_via` column.

Reply:
Ah I see, thanks for clearing that up. I will add the presence validation.